src/famfs_misc.c

```diff
@@ -198,7 +198,8 @@ famfs_get_system_uuid(uuid_le *uuid_out)
 		fprintf(stderr, "%s: unable to read system uuid at %s, errno: %d\n",
 				__func__, sys_uuid_file_path, errno);
 		fclose(f);
-		unlink(sys_uuid_file_path);
+		if (!mock_uuid) // do not delete uuid-file if it is mock test
+			unlink(sys_uuid_file_path);
```

Comment:
I think we can do better here. The condition for this branch is that the uuid (in the sys_uuid_file_path) is invalid. If we're mocking / testing, we should not be here without mocking sys_uuid_file_path.

Above, how about:
```
	char *sys_uuid_dir = SYS_UUID_DIR;

	if (mock_uuid)
		sys_uuid_dir = "/tmp"

	snprintf(sys_uuid_file_path, PATH_MAX - 1, "%s/%s",
			sys_uuid_dir, SYS_UUID_FILE);
```
Then you don't need to test mock_uuid inside the branch, and this function will never screw with the production SYS_UUID_FILE in mocking mode.

I'm sure my pseudo-code above can be improved upon (I didn't compile it), so please check my logic...

Reply:
Done!